adb_shell/adb_device_async.py

```diff
@@ -1097,9 +1097,9 @@ async def _filesync_send(self, command_id, adb_info, filesync_info, data=b'', si
             Optionally override size from len(data).
 
         """
-        if data:
-            if not isinstance(data, bytes):
-                data = data.encode('utf8')
+        if isinstance(data, str) and str is not bytes:
```

Comment:
Since async implies Python3, `str is not bytes` will always be true here.

Reply:
Ah, good point! I'll strip that condition out.